lib/files/node.ts

```diff
@@ -331,31 +352,25 @@ export abstract class Node {
 	/**
 	 * Update the attributes of the node
 	 *
-	 * @param attributes The new attributes to update
+	 * @param attributes The new attributes to update on the Node attributes
 	 */
 	update(attributes: Attribute) {
-		// Proxy the attributes to update the mtime on change
-		// eslint-disable-next-line @typescript-eslint/no-explicit-any
-		this._attributes = new Proxy(this.cleanAttributes(attributes) as any, this.handler)
-	}
-
-	private cleanAttributes(attributes: Attribute): Attribute {
-		const getters = Object.entries(Object.getOwnPropertyDescriptors(Node.prototype))
-			.filter(e => typeof e[1].get === 'function' && e[0] !== '__proto__')
-			.map(e => e[0])
-
-		// Remove protected getters from the attributes
-		// you cannot update them
-		const clean: Attribute = {}
-		for (const key in attributes) {
-			if (getters.includes(key)) {
-				logger.debug(`Discarding protected attribute ${key}`, { node: this, attributes })
-				continue
+		for (const [name, value] of Object.entries(attributes)) {
+			try {
+				if (value === undefined) {
+					delete this.attributes[name]
+				} else {
+					this.attributes[name] = value
+				}
+			} catch (e) {
+				// Ignore readonly attributes
+				if (e instanceof TypeError) {
+					continue
+				}
+				// Throw all other exceptions
+				throw e
```

Comment:
One drawback of this, now there is no way to update ll attributes without updating the mtime :/
That was one of the feature I added and needed from https://github.com/nextcloud-libraries/nextcloud-files/pull/947 :sweat_smile: @susnux

Reply:
Also, you can't remove attributes anymore (unless you know they exist on the node already)

Reply:
https://github.com/nextcloud-libraries/nextcloud-files/pull/973